pkg/v1/tarball/write_test.go

```diff
@@ -484,3 +485,70 @@ func getLayersFilenames(hashes []string) []string {
 	}
 	return filenames
 }
+
+func ExampleWithProgress() {
+	/* calculations for this test:
+	The image we are using is docker.io/library/alpine:3.10
+	its size on disk is 2800640
+	The filesizes inside are:
+	-rw-r--r--  0 0      0        1509 Jan  1  1970 sha256:be4e4bea2c2e15b403bb321562e78ea84b501fb41497472e91ecb41504e8a27c
+	-rw-r--r--  0 0      0     2795580 Jan  1  1970 21c83c5242199776c232920ddb58cfa2a46b17e42ed831ca9001c8dbc532d22d.tar.gz
+	-rw-r--r--  0 0      0         216 Jan  1  1970 manifest.json
+	when rounding each to a 512-byte block, plus the header, we get:
+	1509    ->    1536 + 512 = 2048
+	2795580 -> 2796032 + 512 = 2796544
+	216     ->     512 + 512 = 1024
+	add in 2 blocks of all 0x00 to indicate end of archive
+	                         = 1024
+	                        -------
+	Total:                  2800640
+	*/
+	// buffered channel to make the example test easier
+	c := make(chan v1.Update, 200)
+	// Make a tempfile for tarball writes.
+	fp, err := ioutil.TempFile("", "")
+	if err != nil {
+		fmt.Printf("error creating temp file: %v\n", err)
+		return
+	}
+	defer fp.Close()
+	defer os.Remove(fp.Name())
+
+	tag, err := name.NewDigest("docker.io/library/alpine@sha256:f0e9534a598e501320957059cb2a23774b4d4072e37c7b2cf7e95b241f019e35", name.StrictValidation)
+	if err != nil {
+		fmt.Printf("error creating test tag: %v\n", err)
+		return
+	}
+	desc, err := remote.Get(tag)
+	if err != nil {
+		fmt.Printf("error getting manifest: %v", err)
+		return
+	}
+	img, err := desc.Image()
+	if err != nil {
+		fmt.Printf("error image: %v", err)
+		return
+	}
+	errchan := make(chan error)
+	go func() {
+		if err := tarball.WriteToFile(fp.Name(), tag, img, tarball.WithProgress(c)); err != nil {
+			fmt.Printf("error writing tarball: %v\n", err)
+			errchan <- err
+		}
+		errchan <- nil
+	}()
+	var update v1.Update
+forloop:
+	for {
+		select {
+		case update = <-c:
+			//fmt.Printf("receive update: %#v\n", update)
```

Comment:
Can we uncomment these?

Reply:
Sure, but then I have to print to `os.Stderr`, or it messed up the results.